components/eventing-controller/controllers/subscription/nats/reconciler_test.go

```diff
@@ -262,30 +402,211 @@ func testCreateSubscriptionWithEmptyEventType(id int, _, _ string) bool {
+					v1.ConditionTrue, "")),
+				reconcilertesting.HaveSubsConfiguration(&eventingv1alpha1.SubscriptionConfig{
+					MaxInFlightMessages: defaultSubsConfig.MaxInFlightMessages,
+				}),
+			))
+
+			connection, err := connectToNats(natsURL)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = connection.Publish(natsSubjectToPublish, []byte(reconcilertesting.StructuredCloudEvent))
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// make sure that the subscriber received the message
+			sent := fmt.Sprintf(`"%s"`, reconcilertesting.EventData)
+			Eventually(func() ([]byte, error) {
+				return getFromChanOrTimeout(result, smallPollingInterval)
+			}, timeout, pollingInterval).Should(WithTransform(bytesStringer, Equal(sent)))
 		})
+	})
+}
+
+func testDispatcherWithMultipleSubscribers(id int, eventTypePrefix, natsSubjectToPublish, eventTypeToSubscribe string) bool {
+	return When("Sending Events through Dispatcher for multiple subscribers", func() {
+		It("Should receive events in subscribers", func() {
+			ctx := context.Background()
+
+			var subscription2 *eventingv1alpha1.Subscription
+			eventType := reconcilertesting.EventTypePrefix + "." + reconcilertesting.ApplicationNameNotClean + "." + reconcilertesting.OrderUpdatedV1Event
+			emptyEventType := reconcilertesting.ApplicationNameNotClean + "." + reconcilertesting.OrderUpdatedV1Event
+			natsSubject := reconcilertesting.EventTypePrefix + "." + reconcilertesting.ApplicationName + "." + reconcilertesting.OrderUpdatedV1Event
+			emptyNatsSubject := reconcilertesting.ApplicationName + "." + reconcilertesting.OrderUpdatedV1Event
+
+			// Start reconciler with empty checkSink function
+			cancel = startReconciler(eventTypePrefix, func(ctx context.Context, r *Reconciler, subscription *eventingv1alpha1.Subscription) error {
+				return nil
+			})
+			defer cancel()
+
+			subscriptionName := fmt.Sprintf(subscriptionNameFormat, id)
+
+			subChan1 := make(chan []byte)
+			url, shutdown := newSubscriber(subChan1)
+			defer shutdown()
+
+			subscription := reconcilertesting.NewSubscription(subscriptionName, namespaceName, reconcilertesting.WithFilter(reconcilertesting.EventSource, eventTypeToSubscribe), reconcilertesting.WithWebhookForNats)
+			subscription.Spec.Sink = url
+			ensureSubscriptionCreated(ctx, subscription)
+
+			getSubscription(ctx, subscription).Should(And(
+				reconcilertesting.HaveSubscriptionName(subscriptionName),
+				reconcilertesting.HaveCondition(eventingv1alpha1.MakeCondition(
+					eventingv1alpha1.ConditionSubscriptionActive,
+					eventingv1alpha1.ConditionReasonNATSSubscriptionActive,
+					v1.ConditionTrue, "")),
+				reconcilertesting.HaveSubsConfiguration(&eventingv1alpha1.SubscriptionConfig{
+					MaxInFlightMessages: defaultSubsConfig.MaxInFlightMessages,
+				}),
+			))
+
+			subName2 := fmt.Sprintf("subb-%d", id)
 
-		AfterEach(func() {
-			cancel()
+			subChan2 := make(chan []byte)
+			url2, close := newSubscriber(subChan2)
```

Comment:
Please choose a different name other than `close` as it collides with the builtin close function.